qiskit_dynamics/solver_classes.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+r"""
+=============================================================
+Solver classes (:mod:`qiskit_dynamics.solver_classes`)
+=============================================================
+
+This is temporary, getting docs to build.
+
+.. currentmodule:: qiskit_dynamics.solver_classes
+
+.. autosummary::
+   :toctree: ../stubs/
+
+   Solver
+"""
+
+
+from typing import Optional, Union, Tuple, Any, Type, List
+from copy import deepcopy
+
+import numpy as np
+
+# pylint: disable=unused-import
+from scipy.integrate._ivp.ivp import OdeResult
+
+from qiskit import QiskitError
+
+from qiskit.circuit import Gate, QuantumCircuit
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info.operators.channel.quantum_channel import QuantumChannel
+from qiskit.quantum_info.states.quantum_state import QuantumState
+from qiskit.quantum_info import SuperOp, Operator, Statevector, DensityMatrix
+
+from qiskit_dynamics.models import (
+    BaseGeneratorModel,
+    HamiltonianModel,
+    LindbladModel,
+    RotatingFrame,
+    rotating_wave_approximation,
+)
+from qiskit_dynamics.signals import Signal, SignalList
+from qiskit_dynamics.dispatch import Array
+from qiskit_dynamics import solve_lmde
+
+
+class Solver:
+    """Solver object for simulating both Hamiltonian and Lindblad dynamics.
+    Given the components of a Hamiltonian and optional dissipators, this class will
+    internally construct either a :class:`HamiltonianModel` or :class:`LindbladModel`
+    instance. The evolution given by the model can be simulated by calling :meth:`solve`,
+    which automatically handles :mod:`qiskit.quantum_info` state and super operator types,
+    and calls :func:`~qiskit_dynamics.solve.solve_lmde` to solve.
+    """
+
+    def __init__(
+        self,
+        hamiltonian_operators: Array,
```

Comment:
Yeah makes sense. This will filter down to several places though so let me think about how big a deal it would be to change this - would likely require a separate PR.